common/types/shared.go

```diff
@@ -209,6 +285,64 @@ func (v *ActivityTaskScheduledEventAttributes) GetHeartbeatTimeoutSeconds() (o i
 	return
 }
 
+// Size returns the approximate memory used in bytes
+func (v *ActivityTaskScheduledEventAttributes) Size() uint64 {
+	if v == nil {
+		return 0
+	}
+
+	size := uint64(40)
+
+	if v.ActivityID != "" {
+		size += uint64(16 + len(v.ActivityID))
+	}
+
+	if v.ActivityType != nil {
+		if v.ActivityType.Name != "" {
+			size += uint64(16 + len(v.ActivityType.Name))
+		}
+		size += 8
+	}
+
+	if v.Domain != nil {
+		size += uint64(16 + len(*v.Domain))
+		size += 8
+	}
+
+	if v.TaskList != nil {
+		if v.TaskList.Name != "" {
+			size += uint64(16 + len(v.TaskList.Name))
+		}
+		size += 16
+	}
+
+	if v.Input != nil {
+		size += uint64(len(v.Input))
+	}
+
+	if v.ScheduleToCloseTimeoutSeconds != nil {
+		size += 8
+	}
+	if v.ScheduleToStartTimeoutSeconds != nil {
+		size += 8
+	}
+	if v.StartToCloseTimeoutSeconds != nil {
+		size += 8
+	}
+	if v.HeartbeatTimeoutSeconds != nil {
+		size += 8
+	}
+
+	if v.RetryPolicy != nil {
```

Comment:
RetryPolicy can also implement `Size()`. It shouldn't be hardcoded to 24. It has a string list field